src/Windows/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveWorkflow.cs

```diff
@@ -23,16 +22,14 @@
 using Microsoft.VisualStudio.R.Package.Repl;
 
 namespace Microsoft.R.Components.InteractiveWorkflow.Implementation {
-    public sealed class RInteractiveWorkflow : IRInteractiveWorkflow {
+    public sealed class RInteractiveWorkflow : IRInteractiveWorkflowVisual {
```

Comment:
I spent a day on this and they will be huge. It will take another couple of days + test fixes + review. It entails removal of `GetOrCreateVisualComponent` or equivalent from Workflow, PM, CM, History and Plots. Moving visual-handling code to a separate class and exporting/adding as service. Changing all commands and related code that assume they can get active window etc from workflow, active device from plot manager and so on. Adding test classes (or splitting existing ones), updating tests.